// @textile/hub-threads-client/dist/client.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
const loglevel_1 = __importDefault(require("loglevel"));
const usersd_pb_1 = require("@textile/users-grpc/api/usersd/pb/usersd_pb");
const usersd_pb_service_1 = require("@textile/users-grpc/api/usersd/pb/usersd_pb_service");
const threads_client_1 = require("@textile/threads-client");
Object.defineProperty(exports, "Client", { enumerable: true, get: function () { return threads_client_1.Client; } });
const threads_id_1 = require("@textile/threads-id");
const logger = loglevel_1.default.getLogger('users');
threads_client_1.Client.prototype.getThread = function (name, ctx) {
    return __awaiter(this, void 0, void 0, function* () {
        logger.debug('get thread request');
        const client = new usersd_pb_service_1.APIServiceClient(this.serviceHost, {
            transport: this.rpcOptions.transport,
            debug: this.rpcOptions.debug,
        });
        return new Promise((resolve, reject) => {
            const req = new usersd_pb_1.GetThreadRequest();
            req.setName(name);
            this.context
                .toMetadata(ctx)
                .then((meta) => {
                client.getThread(req, meta, (err, message) => {
                    if (err)
                        reject(err);
                    if (message) {
                        const res = {
                            name: message.getName(),
                            isDb: message.getIsDb(),
                            id: threads_id_1.ThreadID.fromBytes(message.getId_asU8()).toString(),
                        };
                        resolve(res);
                    }
                    else {
                        resolve();
                    }
                });
            })
                .catch((err) => {
                reject(err);
            });
        });
    });
};
/**
 * Returns a list of available Threads.
 * @param ctx Context containing gRPC headers and settings.
 * These will be merged with any internal credentials.
 * @note Threads can be created using the threads or threads network clients.
 */
threads_client_1.Client.prototype.listThreads = function (ctx) {
    return __awaiter(this, void 0, void 0, function* () {
        logger.debug('list threads request');
        const client = new usersd_pb_service_1.APIServiceClient(this.serviceHost, {
            transport: this.rpcOptions.transport,
            debug: this.rpcOptions.debug,
        });
        return new Promise((resolve, reject) => {
            const req = new usersd_pb_1.ListThreadsRequest();
            this.context
                .toMetadata(ctx)
                .then((meta) => {
                client.listThreads(req, meta, (err, message) => {
                    if (err)
                        reject(err);
                    const lst = message === null || message === void 0 ? void 0 : message.getListList();
                    const listList = [];
                    if (lst) {
                        for (let thrd of lst) {
                            const row = thrd.toObject();
                            listList.push(Object.assign(Object.assign({}, row), { id: threads_id_1.ThreadID.fromBytes(thrd.getId_asU8()).toString() }));
                        }
                    }
                    resolve({ listList });
                });
            })
                .catch((err) => {
                reject(err);
            });
        });
    });
};
