// @textile/grpc-powergate-client/dist/powergate/admin/v1/admin_pb.js
// source: powergate/admin/v1/admin.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var powergate_user_v1_user_pb = require('../../../powergate/user/v1/user_pb.js');
goog.object.extend(proto, powergate_user_v1_user_pb);
goog.exportSymbol('proto.powergate.admin.v1.AddressesRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.AddressesResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.CreateUserRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.CreateUserResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.ExecutingStorageJobsRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.ExecutingStorageJobsResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.LatestFinalStorageJobsRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.LatestFinalStorageJobsResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.NewAddressRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.NewAddressResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.QueuedStorageJobsRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.QueuedStorageJobsResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.SendFilRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.SendFilResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.StorageJobsSummaryRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.StorageJobsSummaryResponse', null, global);
goog.exportSymbol('proto.powergate.admin.v1.User', null, global);
goog.exportSymbol('proto.powergate.admin.v1.UsersRequest', null, global);
goog.exportSymbol('proto.powergate.admin.v1.UsersResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.NewAddressRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.NewAddressRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.NewAddressRequest.displayName = 'proto.powergate.admin.v1.NewAddressRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.NewAddressResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.NewAddressResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.NewAddressResponse.displayName = 'proto.powergate.admin.v1.NewAddressResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.AddressesRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.AddressesRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.AddressesRequest.displayName = 'proto.powergate.admin.v1.AddressesRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.AddressesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.AddressesResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.AddressesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.AddressesResponse.displayName = 'proto.powergate.admin.v1.AddressesResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.SendFilRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.SendFilRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.SendFilRequest.displayName = 'proto.powergate.admin.v1.SendFilRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.SendFilResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.SendFilResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.SendFilResponse.displayName = 'proto.powergate.admin.v1.SendFilResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.User = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.User, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.User.displayName = 'proto.powergate.admin.v1.User';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.CreateUserRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.CreateUserRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.CreateUserRequest.displayName = 'proto.powergate.admin.v1.CreateUserRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.CreateUserResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.CreateUserResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.CreateUserResponse.displayName = 'proto.powergate.admin.v1.CreateUserResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.UsersRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.powergate.admin.v1.UsersRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.UsersRequest.displayName = 'proto.powergate.admin.v1.UsersRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.UsersResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.UsersResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.UsersResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.UsersResponse.displayName = 'proto.powergate.admin.v1.UsersResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.QueuedStorageJobsRequest.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.QueuedStorageJobsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.QueuedStorageJobsRequest.displayName = 'proto.powergate.admin.v1.QueuedStorageJobsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.QueuedStorageJobsResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.QueuedStorageJobsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.QueuedStorageJobsResponse.displayName = 'proto.powergate.admin.v1.QueuedStorageJobsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.ExecutingStorageJobsRequest.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.ExecutingStorageJobsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.ExecutingStorageJobsRequest.displayName = 'proto.powergate.admin.v1.ExecutingStorageJobsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.ExecutingStorageJobsResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.ExecutingStorageJobsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.ExecutingStorageJobsResponse.displayName = 'proto.powergate.admin.v1.ExecutingStorageJobsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.LatestFinalStorageJobsRequest.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.LatestFinalStorageJobsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.LatestFinalStorageJobsRequest.displayName = 'proto.powergate.admin.v1.LatestFinalStorageJobsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.LatestFinalStorageJobsResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.LatestFinalStorageJobsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.LatestFinalStorageJobsResponse.displayName = 'proto.powergate.admin.v1.LatestFinalStorageJobsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.displayName = 'proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.displayName = 'proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.StorageJobsSummaryRequest.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.StorageJobsSummaryRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.StorageJobsSummaryRequest.displayName = 'proto.powergate.admin.v1.StorageJobsSummaryRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.powergate.admin.v1.StorageJobsSummaryResponse.repeatedFields_, null);
};
goog.inherits(proto.powergate.admin.v1.StorageJobsSummaryResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.powergate.admin.v1.StorageJobsSummaryResponse.displayName = 'proto.powergate.admin.v1.StorageJobsSummaryResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.NewAddressRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.NewAddressRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.NewAddressRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.NewAddressRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    addressType: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.NewAddressRequest}
 */
proto.powergate.admin.v1.NewAddressRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.NewAddressRequest;
  return proto.powergate.admin.v1.NewAddressRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.NewAddressRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.NewAddressRequest}
 */
proto.powergate.admin.v1.NewAddressRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddressType(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.NewAddressRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.NewAddressRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.NewAddressRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.NewAddressRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAddressType();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string address_type = 1;
 * @return {string}
 */
proto.powergate.admin.v1.NewAddressRequest.prototype.getAddressType = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.NewAddressRequest} returns this
 */
proto.powergate.admin.v1.NewAddressRequest.prototype.setAddressType = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.NewAddressResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.NewAddressResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.NewAddressResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.NewAddressResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    address: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.NewAddressResponse}
 */
proto.powergate.admin.v1.NewAddressResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.NewAddressResponse;
  return proto.powergate.admin.v1.NewAddressResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.NewAddressResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.NewAddressResponse}
 */
proto.powergate.admin.v1.NewAddressResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.NewAddressResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.NewAddressResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.NewAddressResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.NewAddressResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAddress();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string address = 1;
 * @return {string}
 */
proto.powergate.admin.v1.NewAddressResponse.prototype.getAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.NewAddressResponse} returns this
 */
proto.powergate.admin.v1.NewAddressResponse.prototype.setAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.AddressesRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.AddressesRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.AddressesRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.AddressesRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.AddressesRequest}
 */
proto.powergate.admin.v1.AddressesRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.AddressesRequest;
  return proto.powergate.admin.v1.AddressesRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.AddressesRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.AddressesRequest}
 */
proto.powergate.admin.v1.AddressesRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.AddressesRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.AddressesRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.AddressesRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.AddressesRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.AddressesResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.AddressesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.AddressesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.AddressesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.AddressesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    addressesList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.AddressesResponse}
 */
proto.powergate.admin.v1.AddressesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.AddressesResponse;
  return proto.powergate.admin.v1.AddressesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.AddressesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.AddressesResponse}
 */
proto.powergate.admin.v1.AddressesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addAddresses(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.AddressesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.AddressesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.AddressesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.AddressesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAddressesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
};


/**
 * repeated string addresses = 1;
 * @return {!Array<string>}
 */
proto.powergate.admin.v1.AddressesResponse.prototype.getAddressesList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.powergate.admin.v1.AddressesResponse} returns this
 */
proto.powergate.admin.v1.AddressesResponse.prototype.setAddressesList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.AddressesResponse} returns this
 */
proto.powergate.admin.v1.AddressesResponse.prototype.addAddresses = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.AddressesResponse} returns this
 */
proto.powergate.admin.v1.AddressesResponse.prototype.clearAddressesList = function() {
  return this.setAddressesList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.SendFilRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.SendFilRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.SendFilRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.SendFilRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    from: jspb.Message.getFieldWithDefault(msg, 1, ""),
    to: jspb.Message.getFieldWithDefault(msg, 2, ""),
    amount: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.SendFilRequest}
 */
proto.powergate.admin.v1.SendFilRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.SendFilRequest;
  return proto.powergate.admin.v1.SendFilRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.SendFilRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.SendFilRequest}
 */
proto.powergate.admin.v1.SendFilRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setFrom(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setTo(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setAmount(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.SendFilRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.SendFilRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.SendFilRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.SendFilRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getFrom();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getTo();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getAmount();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional string from = 1;
 * @return {string}
 */
proto.powergate.admin.v1.SendFilRequest.prototype.getFrom = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.SendFilRequest} returns this
 */
proto.powergate.admin.v1.SendFilRequest.prototype.setFrom = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string to = 2;
 * @return {string}
 */
proto.powergate.admin.v1.SendFilRequest.prototype.getTo = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.SendFilRequest} returns this
 */
proto.powergate.admin.v1.SendFilRequest.prototype.setTo = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string amount = 3;
 * @return {string}
 */
proto.powergate.admin.v1.SendFilRequest.prototype.getAmount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.SendFilRequest} returns this
 */
proto.powergate.admin.v1.SendFilRequest.prototype.setAmount = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.SendFilResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.SendFilResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.SendFilResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.SendFilResponse.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.SendFilResponse}
 */
proto.powergate.admin.v1.SendFilResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.SendFilResponse;
  return proto.powergate.admin.v1.SendFilResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.SendFilResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.SendFilResponse}
 */
proto.powergate.admin.v1.SendFilResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.SendFilResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.SendFilResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.SendFilResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.SendFilResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.User.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.User.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.User} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.User.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    token: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.User}
 */
proto.powergate.admin.v1.User.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.User;
  return proto.powergate.admin.v1.User.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.User} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.User}
 */
proto.powergate.admin.v1.User.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setToken(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.User.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.User.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.User} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.User.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getToken();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.powergate.admin.v1.User.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.User} returns this
 */
proto.powergate.admin.v1.User.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string token = 2;
 * @return {string}
 */
proto.powergate.admin.v1.User.prototype.getToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.User} returns this
 */
proto.powergate.admin.v1.User.prototype.setToken = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.CreateUserRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.CreateUserRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.CreateUserRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.CreateUserRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.CreateUserRequest}
 */
proto.powergate.admin.v1.CreateUserRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.CreateUserRequest;
  return proto.powergate.admin.v1.CreateUserRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.CreateUserRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.CreateUserRequest}
 */
proto.powergate.admin.v1.CreateUserRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.CreateUserRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.CreateUserRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.CreateUserRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.CreateUserRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.CreateUserResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.CreateUserResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.CreateUserResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.CreateUserResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    user: (f = msg.getUser()) && proto.powergate.admin.v1.User.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.CreateUserResponse}
 */
proto.powergate.admin.v1.CreateUserResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.CreateUserResponse;
  return proto.powergate.admin.v1.CreateUserResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.CreateUserResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.CreateUserResponse}
 */
proto.powergate.admin.v1.CreateUserResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.powergate.admin.v1.User;
      reader.readMessage(value,proto.powergate.admin.v1.User.deserializeBinaryFromReader);
      msg.setUser(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.CreateUserResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.CreateUserResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.CreateUserResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.CreateUserResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUser();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.powergate.admin.v1.User.serializeBinaryToWriter
    );
  }
};


/**
 * optional User user = 1;
 * @return {?proto.powergate.admin.v1.User}
 */
proto.powergate.admin.v1.CreateUserResponse.prototype.getUser = function() {
  return /** @type{?proto.powergate.admin.v1.User} */ (
    jspb.Message.getWrapperField(this, proto.powergate.admin.v1.User, 1));
};


/**
 * @param {?proto.powergate.admin.v1.User|undefined} value
 * @return {!proto.powergate.admin.v1.CreateUserResponse} returns this
*/
proto.powergate.admin.v1.CreateUserResponse.prototype.setUser = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.powergate.admin.v1.CreateUserResponse} returns this
 */
proto.powergate.admin.v1.CreateUserResponse.prototype.clearUser = function() {
  return this.setUser(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.powergate.admin.v1.CreateUserResponse.prototype.hasUser = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.UsersRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.UsersRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.UsersRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.UsersRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.UsersRequest}
 */
proto.powergate.admin.v1.UsersRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.UsersRequest;
  return proto.powergate.admin.v1.UsersRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.UsersRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.UsersRequest}
 */
proto.powergate.admin.v1.UsersRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.UsersRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.UsersRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.UsersRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.UsersRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.UsersResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.UsersResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.UsersResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.UsersResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.UsersResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    usersList: jspb.Message.toObjectList(msg.getUsersList(),
    proto.powergate.admin.v1.User.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.UsersResponse}
 */
proto.powergate.admin.v1.UsersResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.UsersResponse;
  return proto.powergate.admin.v1.UsersResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.UsersResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.UsersResponse}
 */
proto.powergate.admin.v1.UsersResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.powergate.admin.v1.User;
      reader.readMessage(value,proto.powergate.admin.v1.User.deserializeBinaryFromReader);
      msg.addUsers(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.UsersResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.UsersResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.UsersResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.UsersResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUsersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.powergate.admin.v1.User.serializeBinaryToWriter
    );
  }
};


/**
 * repeated User users = 1;
 * @return {!Array<!proto.powergate.admin.v1.User>}
 */
proto.powergate.admin.v1.UsersResponse.prototype.getUsersList = function() {
  return /** @type{!Array<!proto.powergate.admin.v1.User>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.powergate.admin.v1.User, 1));
};


/**
 * @param {!Array<!proto.powergate.admin.v1.User>} value
 * @return {!proto.powergate.admin.v1.UsersResponse} returns this
*/
proto.powergate.admin.v1.UsersResponse.prototype.setUsersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.powergate.admin.v1.User=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.User}
 */
proto.powergate.admin.v1.UsersResponse.prototype.addUsers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.powergate.admin.v1.User, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.UsersResponse} returns this
 */
proto.powergate.admin.v1.UsersResponse.prototype.clearUsersList = function() {
  return this.setUsersList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.QueuedStorageJobsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.QueuedStorageJobsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    userId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    cidsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsRequest}
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.QueuedStorageJobsRequest;
  return proto.powergate.admin.v1.QueuedStorageJobsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.QueuedStorageJobsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsRequest}
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setUserId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addCids(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.QueuedStorageJobsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.QueuedStorageJobsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUserId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCidsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string user_id = 1;
 * @return {string}
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.getUserId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.setUserId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string cids = 2;
 * @return {!Array<string>}
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.getCidsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.setCidsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.addCids = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.QueuedStorageJobsRequest.prototype.clearCidsList = function() {
  return this.setCidsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.QueuedStorageJobsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.QueuedStorageJobsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    storageJobsList: jspb.Message.toObjectList(msg.getStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsResponse}
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.QueuedStorageJobsResponse;
  return proto.powergate.admin.v1.QueuedStorageJobsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.QueuedStorageJobsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsResponse}
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addStorageJobs(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.QueuedStorageJobsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.QueuedStorageJobsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
};


/**
 * repeated powergate.user.v1.StorageJob storage_jobs = 1;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.prototype.getStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 1));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsResponse} returns this
*/
proto.powergate.admin.v1.QueuedStorageJobsResponse.prototype.setStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.prototype.addStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.QueuedStorageJobsResponse} returns this
 */
proto.powergate.admin.v1.QueuedStorageJobsResponse.prototype.clearStorageJobsList = function() {
  return this.setStorageJobsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.ExecutingStorageJobsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    userId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    cidsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsRequest}
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.ExecutingStorageJobsRequest;
  return proto.powergate.admin.v1.ExecutingStorageJobsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsRequest}
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setUserId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addCids(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.ExecutingStorageJobsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUserId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCidsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string user_id = 1;
 * @return {string}
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.getUserId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.setUserId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string cids = 2;
 * @return {!Array<string>}
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.getCidsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.setCidsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.addCids = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.ExecutingStorageJobsRequest.prototype.clearCidsList = function() {
  return this.setCidsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.ExecutingStorageJobsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.ExecutingStorageJobsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    storageJobsList: jspb.Message.toObjectList(msg.getStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsResponse}
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.ExecutingStorageJobsResponse;
  return proto.powergate.admin.v1.ExecutingStorageJobsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.ExecutingStorageJobsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsResponse}
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addStorageJobs(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.ExecutingStorageJobsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.ExecutingStorageJobsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
};


/**
 * repeated powergate.user.v1.StorageJob storage_jobs = 1;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.prototype.getStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 1));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsResponse} returns this
*/
proto.powergate.admin.v1.ExecutingStorageJobsResponse.prototype.setStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.prototype.addStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.ExecutingStorageJobsResponse} returns this
 */
proto.powergate.admin.v1.ExecutingStorageJobsResponse.prototype.clearStorageJobsList = function() {
  return this.setStorageJobsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.LatestFinalStorageJobsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    userId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    cidsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.LatestFinalStorageJobsRequest;
  return proto.powergate.admin.v1.LatestFinalStorageJobsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setUserId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addCids(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.LatestFinalStorageJobsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUserId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCidsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string user_id = 1;
 * @return {string}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.getUserId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.setUserId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string cids = 2;
 * @return {!Array<string>}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.getCidsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.setCidsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.addCids = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestFinalStorageJobsRequest.prototype.clearCidsList = function() {
  return this.setCidsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.LatestFinalStorageJobsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    storageJobsList: jspb.Message.toObjectList(msg.getStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.LatestFinalStorageJobsResponse;
  return proto.powergate.admin.v1.LatestFinalStorageJobsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addStorageJobs(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.LatestFinalStorageJobsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
};


/**
 * repeated powergate.user.v1.StorageJob storage_jobs = 1;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.prototype.getStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 1));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse} returns this
*/
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.prototype.setStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.prototype.addStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.LatestFinalStorageJobsResponse} returns this
 */
proto.powergate.admin.v1.LatestFinalStorageJobsResponse.prototype.clearStorageJobsList = function() {
  return this.setStorageJobsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    userId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    cidsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest;
  return proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setUserId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addCids(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUserId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCidsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string user_id = 1;
 * @return {string}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.getUserId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.setUserId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string cids = 2;
 * @return {!Array<string>}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.getCidsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.setCidsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.addCids = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest} returns this
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsRequest.prototype.clearCidsList = function() {
  return this.setCidsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    storageJobsList: jspb.Message.toObjectList(msg.getStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse;
  return proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addStorageJobs(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
};


/**
 * repeated powergate.user.v1.StorageJob storage_jobs = 1;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.prototype.getStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 1));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse} returns this
*/
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.prototype.setStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.prototype.addStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse} returns this
 */
proto.powergate.admin.v1.LatestSuccessfulStorageJobsResponse.prototype.clearStorageJobsList = function() {
  return this.setStorageJobsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.StorageJobsSummaryRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.StorageJobsSummaryRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    userId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    cidsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryRequest}
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.StorageJobsSummaryRequest;
  return proto.powergate.admin.v1.StorageJobsSummaryRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.StorageJobsSummaryRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryRequest}
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setUserId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addCids(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.StorageJobsSummaryRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.StorageJobsSummaryRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getUserId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCidsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string user_id = 1;
 * @return {string}
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.getUserId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryRequest} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.setUserId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string cids = 2;
 * @return {!Array<string>}
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.getCidsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryRequest} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.setCidsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryRequest} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.addCids = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryRequest} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryRequest.prototype.clearCidsList = function() {
  return this.setCidsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.repeatedFields_ = [2,3,4,5];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.powergate.admin.v1.StorageJobsSummaryResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.powergate.admin.v1.StorageJobsSummaryResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    jobCounts: (f = msg.getJobCounts()) && powergate_user_v1_user_pb.JobCounts.toObject(includeInstance, f),
    queuedStorageJobsList: jspb.Message.toObjectList(msg.getQueuedStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance),
    executingStorageJobsList: jspb.Message.toObjectList(msg.getExecutingStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance),
    latestFinalStorageJobsList: jspb.Message.toObjectList(msg.getLatestFinalStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance),
    latestSuccessfulStorageJobsList: jspb.Message.toObjectList(msg.getLatestSuccessfulStorageJobsList(),
    powergate_user_v1_user_pb.StorageJob.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.powergate.admin.v1.StorageJobsSummaryResponse;
  return proto.powergate.admin.v1.StorageJobsSummaryResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.powergate.admin.v1.StorageJobsSummaryResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new powergate_user_v1_user_pb.JobCounts;
      reader.readMessage(value,powergate_user_v1_user_pb.JobCounts.deserializeBinaryFromReader);
      msg.setJobCounts(value);
      break;
    case 2:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addQueuedStorageJobs(value);
      break;
    case 3:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addExecutingStorageJobs(value);
      break;
    case 4:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addLatestFinalStorageJobs(value);
      break;
    case 5:
      var value = new powergate_user_v1_user_pb.StorageJob;
      reader.readMessage(value,powergate_user_v1_user_pb.StorageJob.deserializeBinaryFromReader);
      msg.addLatestSuccessfulStorageJobs(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.powergate.admin.v1.StorageJobsSummaryResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.powergate.admin.v1.StorageJobsSummaryResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getJobCounts();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      powergate_user_v1_user_pb.JobCounts.serializeBinaryToWriter
    );
  }
  f = message.getQueuedStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
  f = message.getExecutingStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
  f = message.getLatestFinalStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
  f = message.getLatestSuccessfulStorageJobsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      powergate_user_v1_user_pb.StorageJob.serializeBinaryToWriter
    );
  }
};


/**
 * optional powergate.user.v1.JobCounts job_counts = 1;
 * @return {?proto.powergate.user.v1.JobCounts}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.getJobCounts = function() {
  return /** @type{?proto.powergate.user.v1.JobCounts} */ (
    jspb.Message.getWrapperField(this, powergate_user_v1_user_pb.JobCounts, 1));
};


/**
 * @param {?proto.powergate.user.v1.JobCounts|undefined} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
*/
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.setJobCounts = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.clearJobCounts = function() {
  return this.setJobCounts(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.hasJobCounts = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated powergate.user.v1.StorageJob queued_storage_jobs = 2;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.getQueuedStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 2));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
*/
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.setQueuedStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.addQueuedStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.clearQueuedStorageJobsList = function() {
  return this.setQueuedStorageJobsList([]);
};


/**
 * repeated powergate.user.v1.StorageJob executing_storage_jobs = 3;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.getExecutingStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 3));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
*/
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.setExecutingStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.addExecutingStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.clearExecutingStorageJobsList = function() {
  return this.setExecutingStorageJobsList([]);
};


/**
 * repeated powergate.user.v1.StorageJob latest_final_storage_jobs = 4;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.getLatestFinalStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 4));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
*/
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.setLatestFinalStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.addLatestFinalStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.clearLatestFinalStorageJobsList = function() {
  return this.setLatestFinalStorageJobsList([]);
};


/**
 * repeated powergate.user.v1.StorageJob latest_successful_storage_jobs = 5;
 * @return {!Array<!proto.powergate.user.v1.StorageJob>}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.getLatestSuccessfulStorageJobsList = function() {
  return /** @type{!Array<!proto.powergate.user.v1.StorageJob>} */ (
    jspb.Message.getRepeatedWrapperField(this, powergate_user_v1_user_pb.StorageJob, 5));
};


/**
 * @param {!Array<!proto.powergate.user.v1.StorageJob>} value
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
*/
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.setLatestSuccessfulStorageJobsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.powergate.user.v1.StorageJob=} opt_value
 * @param {number=} opt_index
 * @return {!proto.powergate.user.v1.StorageJob}
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.addLatestSuccessfulStorageJobs = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.powergate.user.v1.StorageJob, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.powergate.admin.v1.StorageJobsSummaryResponse} returns this
 */
proto.powergate.admin.v1.StorageJobsSummaryResponse.prototype.clearLatestSuccessfulStorageJobsList = function() {
  return this.setLatestSuccessfulStorageJobsList([]);
};


goog.object.extend(exports, proto.powergate.admin.v1);
