// @textile/grpc-powergate-client/dist/powergate/admin/v1/admin_pb_service.js
// package: powergate.admin.v1
// file: powergate/admin/v1/admin.proto

var powergate_admin_v1_admin_pb = require("../../../powergate/admin/v1/admin_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var AdminService = (function () {
  function AdminService() {}
  AdminService.serviceName = "powergate.admin.v1.AdminService";
  return AdminService;
}());

AdminService.NewAddress = {
  methodName: "NewAddress",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.NewAddressRequest,
  responseType: powergate_admin_v1_admin_pb.NewAddressResponse
};

AdminService.Addresses = {
  methodName: "Addresses",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.AddressesRequest,
  responseType: powergate_admin_v1_admin_pb.AddressesResponse
};

AdminService.SendFil = {
  methodName: "SendFil",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.SendFilRequest,
  responseType: powergate_admin_v1_admin_pb.SendFilResponse
};

AdminService.CreateUser = {
  methodName: "CreateUser",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.CreateUserRequest,
  responseType: powergate_admin_v1_admin_pb.CreateUserResponse
};

AdminService.Users = {
  methodName: "Users",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.UsersRequest,
  responseType: powergate_admin_v1_admin_pb.UsersResponse
};

AdminService.QueuedStorageJobs = {
  methodName: "QueuedStorageJobs",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.QueuedStorageJobsRequest,
  responseType: powergate_admin_v1_admin_pb.QueuedStorageJobsResponse
};

AdminService.ExecutingStorageJobs = {
  methodName: "ExecutingStorageJobs",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.ExecutingStorageJobsRequest,
  responseType: powergate_admin_v1_admin_pb.ExecutingStorageJobsResponse
};

AdminService.LatestFinalStorageJobs = {
  methodName: "LatestFinalStorageJobs",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.LatestFinalStorageJobsRequest,
  responseType: powergate_admin_v1_admin_pb.LatestFinalStorageJobsResponse
};

AdminService.LatestSuccessfulStorageJobs = {
  methodName: "LatestSuccessfulStorageJobs",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.LatestSuccessfulStorageJobsRequest,
  responseType: powergate_admin_v1_admin_pb.LatestSuccessfulStorageJobsResponse
};

AdminService.StorageJobsSummary = {
  methodName: "StorageJobsSummary",
  service: AdminService,
  requestStream: false,
  responseStream: false,
  requestType: powergate_admin_v1_admin_pb.StorageJobsSummaryRequest,
  responseType: powergate_admin_v1_admin_pb.StorageJobsSummaryResponse
};

exports.AdminService = AdminService;

function AdminServiceClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

AdminServiceClient.prototype.newAddress = function newAddress(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.NewAddress, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.addresses = function addresses(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.Addresses, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.sendFil = function sendFil(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.SendFil, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.createUser = function createUser(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.CreateUser, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.users = function users(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.Users, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.queuedStorageJobs = function queuedStorageJobs(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.QueuedStorageJobs, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.executingStorageJobs = function executingStorageJobs(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.ExecutingStorageJobs, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.latestFinalStorageJobs = function latestFinalStorageJobs(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.LatestFinalStorageJobs, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.latestSuccessfulStorageJobs = function latestSuccessfulStorageJobs(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.LatestSuccessfulStorageJobs, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

AdminServiceClient.prototype.storageJobsSummary = function storageJobsSummary(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(AdminService.StorageJobsSummary, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.AdminServiceClient = AdminServiceClient;
